services/users/src/main/java/org/opfab/users/services/UserServiceImp.java

```diff
@@ -104,6 +134,63 @@ public boolean isEachStateUniqueInPerimeter(Perimeter perimeter){
         return true;
     }
 
+    public boolean checkFilteringNotificationIsAllowedForAllProcessesStates(String login, UserSettings userSettings) {
+        if ((userSettings.getProcessesStatesNotNotified() != null) && (!userSettings.getProcessesStatesNotNotified().isEmpty())) {
+            UserData userData = retrieveUser(login);
+
+            if (userData != null) {
+                List<String> groups = userData.getGroups();
+                Set<Perimeter> perimeters = findPerimetersAttachedToGroups(groups);
+
+                CurrentUserWithPerimetersData userWithPerimetersData = CurrentUserWithPerimetersData.builder().userData(userData).build();
+                userWithPerimetersData.computePerimeters(perimeters);
+                if (!isFilteringNotificationAllowedForAllProcessesStates(userWithPerimetersData,
+                        userSettings.getProcessesStatesNotNotified()))
+                    return false;
+            }
+        }
+        return true;
+    }
+
+    private boolean isFilteringNotificationAllowedForAllProcessesStates(CurrentUserWithPerimeters currentUserWithPerimeters,
+                                                                       Map<String, List<String>> processesStates){
+
+        if ((processesStates != null) && (processesStates.size() > 0)) {
+            for (Map.Entry<String, List<String>> entry : processesStates.entrySet()) {
+                List <String> stateIds = entry.getValue();
+                String processId = entry.getKey();
+
+                if (! isFilteringNotificationAllowedForAllProcessStates(currentUserWithPerimeters, processId, stateIds))
+                    return false;
+            }
+        }
+        return true;
+    }
+
+    private boolean isFilteringNotificationAllowedForAllProcessStates(CurrentUserWithPerimeters currentUserWithPerimeters,
+                                                              String processId,
+                                                              List<String> stateIds) {
+        if (stateIds != null) {
+            for (String stateId : stateIds) {
+                if (!isFilteringNotificationAllowedForThisProcessState(currentUserWithPerimeters, processId, stateId))
+                    return false;
+            }
+        }
+        return true;
+    }
+
+    private boolean isFilteringNotificationAllowedForThisProcessState(CurrentUserWithPerimeters currentUserWithPerimeters,
+                                                                     String processId, String stateId){
```

Comment:
proposal for a more efficient code : 

-  Get first all the the filtering not allowed for a user via userPerimeter and put in an object 
-  Go through the settings to check if one is exiting in the object 

To be discuss

Reply:
After discussion with Frederic, this remark will be done in another issue : https://github.com/opfab/operatorfabric-core/issues/3524